### DISCUSSION TITLE
Split web support into a separate package?

### DISCUSSION THREAD

**Discussion by kuhnroyal** [simolus3/drift#2908] [General]
Just want to throw this out here. Is it worth and even doable with the current setup, to split the web support out in a separate package?

Due to the recent WASM support & updates the required dependencies force quiet hard dependency constraints on Flutter projects which are almost impossible to achieve in apps wich have more than a view basic dependencies.
To keep pushing WASM/Web support, the constraints probably need to stay at the most recents dependency versions, which might result in more conflicts in the future.

**Comment by simolus3** [comment]
I'm currently working on factoring some web-specific logic from drift out into a shared package, but since drift would still depend on that package in the future I think it wouldn't help you with the dependencies.

>  To keep pushing WASM/Web support, the constraints probably need to stay at the most recents dependency versions, which might result in more conflicts in the future.

I think the painful gap is the one between packages using the new `dart:js_interop` APIs and the ones continuing to use `package:js` and DOM wrappers in the Dart SDK. Unfortunately, using the new interop form forces us to use version 0.5.0+ of `package:web`. With manageable work, drift could stop depending on `package:js` though if that's causing conflicts for you.
(and I think most packages using version 0.4 of `package:web` should compile against the current one, so maybe a `dependency_override` helps if you don't intend to run your apps on the web either way)

We could add a `drift_wasm` package or something to contain the web-specific parts, but then you'll have the same problem with the `sqlite3` package about to make the same migration. Moving every web-related thing into its own package also increases maintenance work and can be a burden for users interested in web support.
Since this requires a breaking change either way, it's not a short-term solution for web interop. And in the long-term, the dependency issues will hopefully be resolved.

**Reply by kuhnroyal** [reply]
I can probably override as long as there are no compile errors, need to check. Dependabot doesn't like this but it won't be there forever I hope :D

> And in the long-term, the dependency issues will hopefully be resolved.

If that is your expectation, then I am fine with that.
Thanks!
